helix-term/src/commands/typed.rs

```diff
@@ -1473,7 +1473,57 @@ fn tree_sitter_subtree(
             .root_node()
             .descendant_for_byte_range(from, to)
         {
-            let contents = format!("```tsq\n{}\n```", selected_node.to_sexp());
+            fn pretty_print_node(
+                node: Node<'_>,
+                sexpr: &mut String,
+                is_root: bool,
+                field_name: Option<&str>,
+                depth: usize,
+            ) -> anyhow::Result<()> {
```

Comment:
I think we can move this to `helix_core::syntax` and it would be good to add a test case in there. Also, I think this should take a `std::fmt::Write` implementation rather than a String and be split into two functions so it has a friendlier function head:

```rs
pub fn pretty_print_tree<W: std::fmt::Write>(
    fmt: &mut W,
    node: Node,
) -> Result<(), std::fmt::Error> {
    pretty_print_tree_impl(fmt, node, true, None, 0)
}

pub fn pretty_print_tree_impl<W: std::fmt::Write>(
    fmt: &mut W,
    node: Node,
    is_root: bool,
    field_name: Option<&str>,
    depth: usize,
) -> Result<(), std::fmt::Error> {
  // the recursive one
}
```

Reply:
Sounds good! Thanks for the pointer on where to put it :)

Reply:
I moved it to `helix_core::syntax` and added some tests.